testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksUnityCompatibility.java

```diff
@@ -0,0 +1,224 @@
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s order by c1", schemaName, tableName)))
+                    .containsOnly(expectedRowsForUpdate);
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s order by c1", unityCatalogCatalogName, schemaName, tableName)))
+                    .containsOnly(expectedRowsForUpdate);
+
+            // delete
+            List<QueryAssert.Row> expectedRowsForDelete = ImmutableList.of(row(2, "two hundred"));
+            onTrino().executeQuery(format("DELETE FROM delta.%s.%s WHERE c2 = 'one'", schemaName, tableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s order by c1", schemaName, tableName)))
+                    .containsOnly(expectedRowsForDelete);
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s order by c1", unityCatalogCatalogName, schemaName, tableName)))
+                    .containsOnly(expectedRowsForDelete);
+
+            // merge
+            List<QueryAssert.Row> expectedRowsForMerge = ImmutableList.of(row(1, "one"), row(2, "two"), row(3, "three"));
+            String sourceTableName = "test_dt_source_" + randomNameSuffix();
+            String tableLocation2 = getLocationForTable(externalLocationPath, schemaName, sourceTableName);
+            onDelta().executeQuery(
+                    format("CREATE TABLE %s.%s.%s (c1 int, c2 string) using delta location '%s'",
+                            unityCatalogCatalogName, schemaName, sourceTableName, tableLocation2));
+            onDelta().executeQuery(
+                    format("INSERT INTO %s.%s.%s values (1, 'one'), (2, 'two'), (3, 'three')",
+                            unityCatalogCatalogName, schemaName, sourceTableName));
+
+            onTrino().executeQuery(format("MERGE INTO delta.%s.%s t USING delta.%s.%s s on t.c1 = s.c1 " +
+                    "WHEN MATCHED THEN UPDATE SET c2 = s.c2 " +
+                    "WHEN NOT MATCHED THEN INSERT (c1, c2) VALUES (s.c1, s.c2)", schemaName, tableName, schemaName, sourceTableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s order by c1", schemaName, tableName)))
+                    .containsOnly(expectedRowsForMerge);
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s order by c1", unityCatalogCatalogName, schemaName, tableName)))
+                    .containsOnly(expectedRowsForMerge);
+        }
+        finally {
+            onDelta().executeQuery(format("DROP SCHEMA IF EXISTS %s.%s CASCADE", unityCatalogCatalogName, schemaName));
+        }
+    }
+
+    @Test(groups = {DATABRICKS_UNITY_HTTP_HMS, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testTableColumns()
+    {
+        String schemaName = "testCols_" + randomNameSuffix();
```

Comment:
Creating a new schema per test method is redundant. It would be nice to use single schema unless the test purpose requires the dedicated schema.

Reply:
I don't think we need a new schema per test. Can we use `BeforeMethodWithContext` and `AfterMethodWithContext` to create and cleanup the schema? I updated the PR using them. Let me know if that looks okay.